ocaml/xapi/helpers.ml

```diff
@@ -1102,6 +1102,10 @@ let assert_is_valid_cidr kind field cidr =
   if parse_cidr kind cidr = None then
     raise Api_errors.(Server_error (invalid_cidr_address_specified, [field]))
 
+let assert_is_valid_ip_addr kind field address =
+  if not (is_valid_ip kind address || parse_cidr kind address <> None) then
```

Comment:
I think it may be worth trying to do a small ad-hoc method for this.

But on the long term I don't think it's sustainable. Instead, all the IP and CIDR parsing should be done using the ipaddr library, similar to what we did with dates and uuids, we want to stop using ad-hoc methods that use string instead of parsing them to a proper datatype.

Reply:
It seems to me `parse_cidr` does check the 1st part is valid: 
```ocaml
    let address, prefixlen = Scanf.sscanf cidr "%s@/%d" (fun a p -> (a, p)) in
    if not (is_valid_ip kind address) then (
      error "Invalid address in CIDR (%s)" address ;
      None
    )
```

Reply:
Ipaddr.Prefix.of_string seems capable of parsing/validating CIDR, we could use that.